### PR TITLE
Add action item and list, component design, logic, and tests

### PR DESCRIPTION
**Pull request by jaypontes**

### CODE REVIEW

File: services/frontend-service/src/ui/components/SideBar/SideBar.tsx [hunk 1]
```diff
@@ -15,11 +15,167 @@ along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
...
+                dialogTitle: 'Are you sure you want to add this environment lock?',
+                summary: 'Create new environment lock on ' + action.createEnvironmentLock.environment,
+                environment: action.createEnvironmentLock.environment,
+            };
+        case 'deleteEnvironmentLock':
+            return {
+                type: ActionTypes.DeleteEnvironmentLock,
+                name: 'Delete Env Lock',
+                dialogTitle: 'Are you sure you want to delete this environment lock?',
+                summary: 'Delete environment lock on ' + action.deleteEnvironmentLock.environment,
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/SideBar/SideBar.tsx:65]: please add the message of the lock here to the summary
'Delete environement lock (' + action.message + .... on ...

File: services/frontend-service/src/ui/components/SideBar/SideBar.tsx [hunk 2]
```diff
@@ -15,11 +15,167 @@ along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
...
+                application: action.createEnvironmentApplicationLock.application,
+            };
+        case 'deleteEnvironmentApplicationLock':
+            return {
+                type: ActionTypes.DeleteApplicationLock,
+                name: 'Delete App Lock',
+                dialogTitle: 'Are you sure you want to delete this application lock?',
+                summary:
+                    'Unlock "' +
+                    action.deleteEnvironmentApplicationLock.application +
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/SideBar/SideBar.tsx:89]: please add the message of the lock here to the summary
'Delete environement lock (' + action.message + .... on ...

File: services/frontend-service/src/ui/components/SideBar/SideBar.tsx [hunk 3]
```diff
@@ -15,11 +15,167 @@ along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
...
+                    action.deleteEnvironmentApplicationLock.environment,
+                environment: action.deleteEnvironmentApplicationLock.environment,
+                application: action.deleteEnvironmentApplicationLock.application,
+                lockId: action.deleteEnvironmentApplicationLock.lockId,
+            };
+        case 'deploy':
+            return {
+                type: ActionTypes.Deploy,
+                name: 'Deploy',
+                dialogTitle: 'Please be aware:',
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/SideBar/SideBar.tsx:100]:
```suggestion
                dialogTitle: 'Are you sure you want to deploy?',
```

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: services/frontend-service/src/ui/components/SideBar/SideBar.tsx (hunk 3 above)
Comment (jaypontes) [reply services/frontend-service/src/ui/components/SideBar/SideBar.tsx:100]: Done!

**Pull request merged**